oraclizeAPI_0.5.sol

```diff
@@ -77,760 +73,881 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+        return oraclize.query_withGasLimit.value(price)(_timestamp, _datasource, _arg, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string memory _arg, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource, _gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+           return 0; // Unexpectedly high price
+        }
+        return oraclize.query_withGasLimit.value(price)(0, _datasource, _arg, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string memory _arg1, string memory _arg2) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2.value(price)(0, _datasource, _arg1, _arg2);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, string memory _arg1, string memory _arg2) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2.value(price)(_timestamp, _datasource, _arg1, _arg2);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, string memory _arg1, string memory _arg2, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource, _gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2_withGasLimit.value(price)(_timestamp, _datasource, _arg1, _arg2, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string memory _arg1, string memory _arg2, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource, _gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2_withGasLimit.value(price)(0, _datasource, _arg1, _arg2, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string [] memory _argN) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price
+        }
+        bytes memory args = stra2cbor(_argN);
+        return oraclize.queryN.value(price)(0, _datasource, args);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, string [] memory _argN) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price

```

Comment:
i doubt anyone handles this currently, so in a future PR (not this one as here we're standardizing and bringing up to 0.5 spec) might be worth to replace with revert containing message.

Reply:
I've never seen this handled either. Can replace with revert if @bertani and yourself think it's better?